src/plugins/mvcc_delete_plugin.hpp

```diff
@@ -0,0 +1,51 @@
+#pragma once
+
+#include <algorithm>
+#include <mutex>
+#include <queue>
+#include <thread>
+
+#include "gtest/gtest_prod.h"
+#include "storage/chunk.hpp"
+#include "storage/storage_manager.hpp"
+#include "utils/abstract_plugin.hpp"
+#include "utils/pausable_loop_thread.hpp"
+#include "utils/singleton.hpp"
+
+namespace opossum {
+
+class MvccDeletePlugin : public AbstractPlugin, public Singleton<MvccDeletePlugin> {
+  friend class MvccDeletePluginTest;
+
+ public:
+  const std::string description() const final;
+
+  void start() final;
+
+  void stop() final;
+
+ private:
+  struct ChunkSpecifier {
```

Comment:
I do not like names that I need to lookup in order to know what they contain. `using TableAndChunkID = std::pair<std::shared_ptr<Table>, ChunkID>` would sound a bit awkward, but would be more speaking.

Reply:
Thanks. Good suggestion!